codec/src/main/java/io/netty/handler/codec/DefaultTextHeaders.java

```diff
@@ -396,4 +401,188 @@ public TextHeaders clear() {
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, CharSequence... values) {
+            DefaultTextHeaders.super.add(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, Iterable<? extends CharSequence> values) {
+            DefaultTextHeaders.super.add(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Object value) {
+            DefaultTextHeaders.super.addObject(name, commaSeparated(value));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Iterable<?> values) {
+            DefaultTextHeaders.super.addObject(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Object... values) {
+            DefaultTextHeaders.super.addObject(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, CharSequence... values) {
+            DefaultTextHeaders.super.set(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, Iterable<? extends CharSequence> values) {
+            DefaultTextHeaders.super.set(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Object... values) {
+            DefaultTextHeaders.super.setObject(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Iterable<?> values) {
+            DefaultTextHeaders.super.setObject(name, commaSeparated(values));
+            return DefaultTextHeaders.this;
+        }
+
+        private CharSequence commaSeparated(Object... values) {
+            StringBuilder singleValue = new StringBuilder(values.length * DEFAULT_VALUE_SIZE);
+            if (values.length > 0) {
+                int end = values.length - 1;
+                for (int i = 0 ; i < end; i++) {
+                    singleValue.append(values[i]).append(", ");
+                }
+                singleValue.append(values[end]);
+            }
+            return singleValue.toString();
+        }
+
+        private CharSequence commaSeparated(Iterable<?> values) {
+            StringBuilder singleValue = new StringBuilder();
+            Iterator<?> iterator = values.iterator();
+            while (iterator.hasNext()) {
+                singleValue.append(iterator.next());
+                if (iterator.hasNext()) {
```

Comment:
Note...If we expect/allow `null` values this will prematurely exit in its current form (this can be avoided if needed).